### PR TITLE
Introduce components

### PR DESCRIPTION
**Pull request by sottar**
Added ui components package to this repository.

Implements base settings for components package.

### CODE REVIEW

File: .eslintrc.json [hunk 1]
```diff
@@ -1,3 +1,31 @@
...
+    }
+  },
+  "env": {
+    "browser": true,
+    "es6": true,
+    "commonjs": true
+  },
+  "globals": {
+    "process": "readonly"
+  }
```
Comment (9renpoto) [inline .eslintrc.json:30]: can u move https://github.com/caddijp/frontend/blob/master/packages/eslint-config/index.js ?

File: packages/components/tsconfig.json [hunk 2]
```diff
@@ -0,0 +1,70 @@
+{
+  "compilerOptions": {
+    /* Visit https://aka.ms/tsconfig.json to read more about this file */
+
+    /* Basic Options */
+    // "incremental": true,                   /* Enable incremental compilation */
+    "target": "es5" /* Specify ECMAScript target version: 'ES3' (default), 'ES5', 'ES2015', 'ES2016', 'ES2017', 'ES2018', 'ES2019', 'ES2020', or 'ESNEXT'. */,
```
[the remark this answers is not in the document]
Comment (9renpoto) [reply packages/components/tsconfig.json:7]: https://github.com/caddijp/frontend/issues/3 I'll commit super tsconfig base!

File: .prettierrc [hunk 3]
```diff
@@ -0,0 +1,6 @@
+{
+  "singleQuote": true,
+  "trailingComma": "all",
```
Comment (9renpoto) [inline .prettierrc:3]:
```suggestion
```

Do u want this option to use the default  ?

File: .prettierrc [hunk 4]
```diff
@@ -0,0 +1,6 @@
...
+  "semi": true,
```
Comment (9renpoto) [inline .prettierrc:4]:
```suggestion
```

Do u want this option to use the default ?

https://prettier.io/docs/en/options.html#semicolons

File: packages/components/package.json [hunk 5]
```diff
@@ -0,0 +1,31 @@
...
+    "prepublishOnly": "run-s clean lint build",
+    "test": "echo \"Error: no test specified\" && exit 1"
+  },
+  "description": "CADDi ui components built with React.",
+  "repository": "git+https://github.com/caddijp/caddi-components.git",
+  "devDependencies": {
+    "@types/react": "^16.9.46",
+    "@types/react-dom": "^16.9.8",
+    "@types/styled-components": "^5.1.2",
+    "npm-run-all": "^4.1.5",
```
Comment (9renpoto) [inline packages/components/package.json:20]:
```suggestion
```

already installed. please delete it :) https://github.com/caddijp/frontend/blob/2575554735e0ecb8b1168d442c33569a65333c25/package.json#L14

File: packages/components/package.json [hunk 6]
```diff
@@ -0,0 +1,31 @@
...
+    "react": "^16.13.1",
+    "react-dom": "^16.13.1",
+    "styled-components": "^5.1.1",
+    "typescript": "^3.9.7"
```
Comment (9renpoto) [inline packages/components/package.json:24]:
```suggestion
```

already installed. please delete it :)

File: packages/components/package.json [hunk 7]
```diff
@@ -0,0 +1,31 @@
+{
+  "name": "@caddijp/components",
+  "version": "0.0.1",
+  "main": "lib/index.js",
+  "module": "esm/index.js",
```
Comment (9renpoto) [inline packages/components/package.json:5]: 💪

File: packages/components/src/components/Button/Button.tsx [hunk 8]
```diff
@@ -0,0 +1,12 @@
+import React from 'react';
+
+type ButtonProps = {
+  text: string;
+};
+
+const Button: React.FC<ButtonProps> = ({ text }: ButtonProps) => {
```
[the remark this answers is not in the document]
Comment (sottar) [reply packages/components/src/components/Button/Button.tsx:7]: これ eslint のエラー出たんですよね。。。
って思ったけど今やったら通るな。。
あざます 🙏

**Pull request merged**